packages/js/product-editor/src/hooks/use-conditional-execution.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { RefObject } from 'react';
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useVisibilityObserver } from './use-visibility-observer';
+
+type useConditionalExecutionProps = {
+	elementOrSelector?: RefObject< HTMLInputElement > | string;
+	onVisible?: () => void;
+	onHidden?: () => void;
+	isMemorized?: boolean;
+};
+
+export function useConditionalExecution( {
+	elementOrSelector,
+	onVisible,
+	onHidden,
+	isMemorized = false,
+}: useConditionalExecutionProps ) {
+	const elementRef = useRef( null );
+	const isVisible = useVisibilityObserver( elementOrSelector ?? elementRef );
+
+	function getDependencies() {
+		const dependenciesList = [];
+		if ( isMemorized ) {
```

Comment:
This does not work if you want only one function to be cached and not the other. So this always consider both.

Reply:
In this case I prefer to give that responsibility to the consumer instead. Like so `useConditionalExecution({ onVisible: useCallback(cb, deps) })`. What do you thing @louwie17 ?

Reply:
If we decide to do that, every non-memorized function we use as a dependency for the `useEffect` will trigger an infinite loop. I'm ok with delegate that responsibility to the consumer but I would warn in a comment about that situation.

Reply:
I changed that in the commits [3595f50](https://github.com/woocommerce/woocommerce/pull/47025/commits/3595f506919d7a26cbd9170523a36436a3f0b3c0) and [161f3fd](https://github.com/woocommerce/woocommerce/pull/47025/commits/161f3fd343462e9b7788ffeae941d7767f6c8d55)